db/migrate/20251215174251_create_roles.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+
+class CreateRoles < ActiveRecord::Migration[8.0]
+  def up
+    create_table :roles, id: :uuid do |t|
+      t.string :name, null: false
+      t.boolean :admin, null: false, default: false
+      t.text :permissions, array: true, null: false, default: []
+      t.references :organization, type: :uuid
+      t.text :description
+      t.datetime :deleted_at
+      t.timestamps
+
+      t.check_constraint "name <> ''", name: "name_not_empty"
+      t.check_constraint "NOT admin OR permissions = '{}'", name: "admin_cannot_have_permissions"
+      t.check_constraint "admin OR permissions <> '{}'", name: "non_admin_must_have_permissions"
+      t.check_constraint "NOT ('' = ANY(permissions))", name: "permissions_must_not_be_empty"
+      t.check_constraint "NOT (permissions::text ~ '([\\{,]:|::|:[,\\}])')", name: "permissions_has_no_empty_parts"
+
+      t.index [:id, :admin],
+        unique: true,
+        name: "index_roles_by_id_and_admin"
+      t.index :admin,
+        unique: true,
+        where: "admin AND deleted_at IS NULL",
+        name: "index_roles_by_unique_admin"
```

Comment:
So we want only one global role with `admin = true` ?

Reply:
yes, it is required by the spec